### PR TITLE
[Commands]: Fix crash opening a new guest window with the #quick-commands flag enabled

### PR DESCRIPTION
**Pull request by fallaciousreasoning**
Resolves https://github.com/brave/brave-browser/issues/31997

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [x] New files have MPL-2.0 license header
- [x] Adequate test coverage exists to prevent regressions
- [x] Major classes, functions and non-trivial code blocks are well-commented
- [x] Changes in component dependencies are properly reflected in `gn`
- [x] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [x] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
See https://github.com/brave/brave-browser/issues/31997

### CODE REVIEW

**Review by bsclifton — APPROVED** [review]
Changes LGTM - nice cleanup. I pulled the code locally / built / verified the fix 👍

File: components/commander/common/constants.cc [hunk 1]
```diff
@@ -5,8 +5,6 @@
 
 #include "brave/components/commander/common/constants.h"
 
-#include "base/strings/string_piece.h"
-
```
Comment (sangwoo108) [inline components/commander/common/constants.cc:9]: I'm curious why this was removed.

**Review by sangwoo108 — APPROVED** [review]
Nice clean-up! 👍

**Review by iefremov — APPROVED** [review]
chromium_src lgtm

**Comment by iefremov** [comment]
personally i prefer to separate code that fixes the crash from any refactorings, in case the fix would be uplifted somewhere. And for simplicity

**Comment by bsclifton** [comment]
iOS CI is failing 🤔 
```
[2023-08-01T07:08:56.559Z] FAILED: obj/components/omnibox/browser/libbrowser.a 

[2023-08-01T07:08:56.559Z] rm -f obj/components/omnibox/browser/libbrowser.a && "../../third_party/llvm-build/Release+Asserts/bin/llvm-ar"  -r -c -s -D obj/components/omnibox/browser/libbrowser.a @obj/components/omnibox/browser/libbrowser.a.rsp

[2023-08-01T07:08:56.559Z] ../../third_party/llvm-build/Release+Asserts/bin/llvm-ar: error: obj/components/omnibox/browser/browser/omnibox_action.o: No such file or directory
```

File: browser/ui/commander/commander_service_factory.cc [hunk 2]
```diff
@@ -34,6 +35,8 @@ CommanderServiceFactory::CommanderServiceFactory()
           "CommanderService",
           ProfileSelections::Builder()
               .WithRegular(ProfileSelection::kOwnInstance)
+              .WithGuest(ProfileSelection::kOwnInstance)
+              .WithSystem(ProfileSelection::kOwnInstance)
```
Comment (bridiver) [inline browser/ui/commander/commander_service_factory.cc:39]: I think this should be kNone, no?

File: components/commander/common/constants.cc (hunk 1 above)
Comment (bridiver) [reply components/commander/common/constants.cc:9]: it's in the header probably

**Review by bridiver — APPROVED** [review]

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 11 x64` using the following build(s):

```
Brave | 1.58.80 Chromium: 116.0.5845.51 (Official Build) nightly (64-bit)
-- | --
Revision | 4408cc9795df260c5d5f033660ce757782b53b32
OS | Windows 11 Version 22H2 (Build 22621.1992)
```

Using the STR/Cases outlined via https://github.com/brave/brave-browser/issues/31997#issue-1830113820 using `1.58.52 Chromium: 116.0.5845.51`, reproduced the crash when opening a `Guest Window` after enabling `brave://flags/#quick-commands` via `brave://flags`. Using `1.58.80 Chromium: 116.0.5845.51`, ensured that the crash wasn't occurring.

**Pull request merged**